crates/core/src/delta_datafusion/mod.rs

```diff
@@ -2388,4 +2442,86 @@ mod tests {
             self.inner.rename_if_not_exists(from, to).await
         }
     }
+
+    #[test]
+    fn test_find_files_tracing_span() {
+        let span = tracing::info_span!(
+            "find_files",
+            version = 10,
+            has_predicate = true,
+            partition_scan = tracing::field::Empty,
+            candidate_count = tracing::field::Empty
+        );
+
+        let metadata = span.metadata().expect("span should have metadata");
+        assert_eq!(metadata.name(), "find_files");
+        assert_eq!(metadata.level(), &tracing::Level::INFO);
+        assert!(metadata.is_span());
+
+        span.record("partition_scan", true);
+        span.record("candidate_count", 45);
+    }
+
+    #[test]
+    fn test_find_files_scan_tracing_span() {
+        let span = tracing::info_span!(
+            "find_files_scan",
+            version = 10,
+            total_files = tracing::field::Empty,
+            matching_files = tracing::field::Empty
+        );
+
+        let metadata = span.metadata().expect("span should have metadata");
+        assert_eq!(metadata.name(), "find_files_scan");
+        assert_eq!(metadata.level(), &tracing::Level::INFO);
+
+        span.record("total_files", 1000);
+        span.record("matching_files", 12);
+    }
+
+    #[test]
+    fn test_find_files_tracing_log_events() {
+        tracing::debug!("using partition-only scan (memory table)");
+        tracing::debug!("using physical data scan");
+        tracing::debug!("no predicate, returning all files");
+
+        tracing::info!(
+            candidates = 45,
+            scan_type = "partition",
+            "file pruning completed"
+        );
+
+        tracing::info!(
+            candidates = 12,
+            scan_type = "physical",
+            "file pruning completed"
+        );
+
+        tracing::debug!(matching_files = 12, "physical scan completed");
+    }
```

Comment:
:laughing: this test doesn't actually do anything

Reply:
I'm working on it... I removed some examples where I tested the execution.
I wanted to see the test coverage results first.